crates/bevy_reflect/src/reflect.rs

```diff
@@ -46,17 +49,13 @@ pub enum ReflectMut<'a> {
 ///
 /// When using `#[derive(Reflect)]` with a struct or tuple struct, the suitable subtrait for that
 /// type (`Struct` or `TupleStruct`) is derived automatically.
-///
-/// # Safety
-/// Implementors _must_ ensure that [`Reflect::any`] and [`Reflect::any_mut`] both return the `self`
-/// value passed in. If this is not done, [`Reflect::downcast`](trait.Reflect.html#method.downcast)
-/// will be UB (and also just logically broken).
-pub unsafe trait Reflect: Any + Send + Sync {
-    /// Returns the [type name] of the underlying type.
-    ///
-    /// [type name]: std::any::type_name
+pub trait Reflect: Any + Send + Sync {
+    /// Returns the [Type Name][std::any::type_name] of the underlying type.
```

Comment:
I don't know what this is supposed to look like in docs, but for me this looked like:
> Returns the [name] of the underlying type
It looks very much unintentional. I think the `type` gets removed for some reason? How about type_name instead?

Reply:
Strange it looks fine for me (CLion). What do the other instances of "type name" look like for you on other functions (`***Info::type_name` for example)?

Reply:
ValueInfo also looks like this:
![image](https://user-images.githubusercontent.com/42193387/174334272-7ce0458d-d647-499f-9403-232e103cc4fc.png)
`type` gets removed, and it seems like it's supposed to be a link but that style does not work. The [name][link] style does work.

Reply:
Huh. Looks like a bug since that's valid markdown. I lean more towards keeping it "type name" and let RA fix that issue at some point (it should work at least for docs.rs). 

But if you want to do "name" or "type_name" we can always come back in the future to do a bit of doc cleanup haha

Reply:
Keeping it is fine by me; I just didn't realize it was valid.